### PR TITLE
Standardize memory storage

### PR DESCRIPTION
**Pull request by scottinet**
# Description

Make the `MemoryStorage` object on par with the newly refactored corresponding controller and with the new SDK reference.

# Other changes

* the `Kuzzle.query` function now handles any querystring parameter other than the preset list

### CODE REVIEW

File: src/MemoryStorage.php [hunk 1]
```diff
@@ -261,53 +292,314 @@ public function __construct(Kuzzle $kuzzle)
...
-        ];
-
-        foreach ($this->COMMANDS[$command] as $key => $value) {
-            if (!array_key_exists($key, $arguments)) {
-                continue;
+        if ($argsLeft === 1 && !is_array($arguments[0])) {
+            throw new InvalidArgumentException('MemoryStorage.' . $command . ': Invalid optional parameter (expected an associative array)');
+        }
+
+        if ($argsLeft == 1) {
```
Comment (dbengsch) [inline src/MemoryStorage.php]: `===` instead of `==`

**Review by dbengsch — CHANGES_REQUESTED** [review]
Nice work !

File: src/MemoryStorage.php (hunk 1 above)
Comment (scottinet) [reply src/MemoryStorage.php]: fixed

**Review by dbengsch — APPROVED** [review]

File: src/MemoryStorage.php [hunk 2]
```diff
@@ -261,53 +292,314 @@ public function __construct(Kuzzle $kuzzle)
 
     public function __call($command, $arguments)
     {
-        if (!array_key_exists($command, $this->COMMANDS)) {
+        if (!isset($this->COMMANDS[$command])) {
```
Comment (stafyniaksacha) [inline src/MemoryStorage.php:295]: usage of `array_key_exists` is required, `isset` will throw an error if there are no `$command` key into `$this->COMMANDS` array

**Review by stafyniaksacha — CHANGES_REQUESTED** [review]

File: src/MemoryStorage.php (hunk 2 above)
Comment (scottinet) [reply src/MemoryStorage.php:295]: That does not seem to be the case according to [PHP documentation](http://php.net/manual/en/function.isset.php), where it is stated that the only difference between these two methods, is that `isset` returns FALSE if an element is set but equals to NULL, where in that specific case, `array_key_exists` will return TRUE.

Other than that, using `isset` on an unexisting element of an array should simply return FALSE.

**Review by stafyniaksacha — APPROVED** [review]

**Pull request merged**